include/nuttx/queue.h

```diff
@@ -261,14 +261,16 @@
       else \
         { \
           (q2)->tail = (q1)->tail; \
-          (q2)->head = p; \
-          (q1)->tail = p->blink; \
-          p->blink = NULL; \
+          (q2)->head = (p); \
```

Comment:
Let's change `if (dq_peek(q1) == p)` to `if (dq_peek(q1) == (p))` few lines above as well